tex/appendix/ans-fo-semantics.tex

```diff
@@ -179,6 +179,24 @@ \subsection*{9.7.6}
     \emptyset$ is valid in our model.
 \end{enumerate}
 
+\subsection*{9.7.10}
+
+\begin{enumerate}
+    \item[i] We will prove $\mathcal{M} \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$ iff $P^{\mathcal{M}}$ has at least two elements. \\\\
+    Left-to right: We will prove this using proof by contradiction. Let $\mathcal{M}$ and  $\alpha$ be arbitrary such that $\mathcal{M}, \alpha \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. Assume $P^\mathcal{M}$ has at most 1 element. We can distinguish two cases: $P^\mathcal{M}$ has 0 elements (1) and $P^\mathcal{M}$ has 1 element (2). \\ 
+    (1) We assume $P^\mathcal{M} = \emptyset$. Then there is no $d \in D^\mathcal{M}$ such that $P(d)$ is true. This means that the conjunction $P(x) \wedge P(y) \wedge x \neq y$ can never be true. But we assumed $\mathcal{M}, \alpha \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. Thus, we have a contradiction. It follows that if $P^\mathcal{M} = \emptyset$, $\mathcal{M}, \alpha \nvDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. \\
+    (2) We assume $P^\mathcal{M}$ has 1 element. Let this be an arbitrary element $d \in D^\mathcal{M}$. In order for $\mathcal{M}, \alpha \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$ to hold, it must be the case that $\mathcal{M}, \alpha_{[x \mapsto d, y \mapsto d]} \vDash (P(d) \wedge P(d) \wedge d \neq d)$. Clearly, $P(d) \wedge P(d)$ holds, but we know $d = d$. Thus, we have a contradiction. It follows that if $P^\mathcal{M}$ has one element, $\mathcal{M}, \alpha \nvDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. \\\\
+    Right-to-left: We will prove this using proof by contradiction. We assume $P^\mathcal{M}$ has at least two elements and $\mathcal{M}, \alpha \nvDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. The latter can be rewritten:
+    \begin{align*}
+        \mathcal{M}, \alpha &\nvDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y) \\
+        &\vDash \neg \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y) \\
+        &\vDash \forall x \neg \exists y (P(x) \wedge P(y) \wedge x \neq y) \\
+        &\vDash \forall x \forall y \neg (P(x) \wedge P(y) \wedge x \neq y) \\
+        &\vDash \forall x \forall y (\neg P(x) \vee \neg P(y) \vee x = y) \\
+    \end{align*} 
+    So, we have $\mathcal{M}, \alpha \vDash \forall x \forall y (\neg P(x) \vee \neg P(y) \vee x = y)$ and we know $P^\mathcal{M}$ has at least two elements. We will give a counterexample. Assume $p, q \in D^\mathcal{M}$ and $P^\mathcal{M} = {p, q}$. It follows that, in order for our claim to be valid, $\mathcal{M}, \alpha_{[x \mapsto p, y \mapsto q} \vDash (\neg P(p) \vee \neg P(q) \vee p = q)$ must hold. However, we know $P(p) \wedge P(q) \wedge p \neq q$. Thus, we have found a counterexample. It follows that if $P^{\mathcal{M}}$ has at least two elements then $\mathcal{M} \vDash \exists x \exists y (P(x) \wedge P(y) \wedge x \neq y)$. $\square$
```

Comment:
It's not ideal to say that $P^\mathcal{M}=p,q$. 1) $P^\mathcal{M}$ is a set, so it should be $P^\mathcal{M}=\{p,q\}$ (that's just notation). 2) You want to assume that $P^\mathcal{M}$ has _at least_ two elements, but this means that you have _exactly_ two elements. You should write something like $p\neq q\in P^\mathcal{M}$. 
I like the idea of how you reason to what it means for $\exists x\exists y(P(x)\land P(y)\land x\neq y)$ to be false under a given assignment: for any objects d and d', either $\mathcal{M},\alpha[x\mapsto d, x\mapsto d']\nvDash P(x)$ or $\mathcal{M},\alpha[x\mapsto d, x\mapsto d']\nvDash P(y)$ or $\mathcal{M},\alpha[x\mapsto d, x\mapsto d']\nvDash x\neq y$. But you go a bit quick at the end: it'd be great if you could just mention that these are the three possible ways in which the formula can be false (via your equivalence reasoning) and that in each case you get a contradiction for your two object.